docs/configuration.md

```diff
@@ -556,6 +576,30 @@ value: <tmpl_string>
 [ short: <boolean> | default = slack_config.short_fields ]
 ```
 
+## `<matcher>`
+
+A matcher is a string with a syntax inspired by PromQL and OpenMetrics. The syntax of a matcher consists of three tokens: 
+
+- A valid Prometheus label name. 
+
+- One of  `=`, `!=`, `=~`, or `!~`. `=` means equals, `!=` means that the strings are not equal, `=~` is used for equality of regex expressions and `!~` is used for un-equality of regex expressions. They have the same meaning as known from PromQL selectors.
+
+- A UTF-8 string, which may be enclosed in double quotes. Before or after each token, there may be any amount of whitespace. 
+
+The 3rd token may be the empty string. Within the 3rd token, OpenMetrics escaping rules apply: `\"` for a double-quote, `\n` for a line feed, `\\` for a literal backslash. Unescaped `"` must not occur inside the 3rd token (only as the 1st or last character). However, literal line feed characters are tolerated, as are single `\` characters not followed by `\`, `n`, or `"`. They act as a literal backslash in that case.
+
+In the configuration, multiple matchers are combined in a YAML list. However, it is also possible to combine multiple matchers within a single YAML string, again using syntax inspired by PromQL. In such a string, a leading { and/or a trailing } is optional and will be trimmed before further parsing. Individual matchers are separated by commas outside of quoted parts of the string. Those commas may be surrounded by whitespace. Parts of the string inside unescaped double quotes `"…"` are considered quoted (and commas don't act as separators there). If double quotes are escaped with a single backslash `\`, they are ignored for the purpose of identifying quoted parts of the input string. If the input string, after trimming the optional trailing }, ends with a comma, followed by optional whitespace, this comma and whitespace will be trimmed.
+
+All of the following is therefore valid:
+
+``` yaml
+matchers: [ '{foo = "bar", dings != "bums", }' ]
+matchers: [ foo=bar,dings!=bums ]
+matchers: [ foo=bar, dings!=bums ]
+matchers: [ '{quote="She said: \"Hi, ladies! That's gender-neutral…\""}' ]
+matchers: [ statuscode=~"5.." ]
+```
```

Comment:
Oh noes. I remember I wrote a whole lot about this section with a lot of examples, that I all tried out, but somehow GitHub has eaten it.

 The problem here is that a short-form list in Yaml uses other string quoting than a long-form list. Also, the above is technically not valid Yaml because it reuses the same key "matchers" in a map. I think we should have several separate examples and explanations in between.

I'll try to recreate what I wrote last week:

Two matchers combined in a long-form YAML list:

```yaml
matchers:
  - foo = bar
  - dings != bums
```

The same in a short-form YAML list:

```yaml
matchers: [ foo = bar, dings != bums ]
```

In the short-form, it's generally better to quote the list elements to avoid problems with special characters like commas:

```yaml
matchers: [ "foo = bar,baz", "dings != bums" ]
```

Or you put both matchers into one PromQL-like string. Single quotes for the whole string work best here:

```yaml
matchers: [ '{foo="bar,baz", dings!="bums"}' ]
```

To avoid any confusion about YAML string quoting and escaping, you can use YAML block quoting and then only worry about the OpenMetrics escaping inside the block. A complex example with a regular expression and different quotes inside the label value:

```yaml
matchers:
  - |
    {quote=~"She said: \"Hi, ladies!( That's gender-neutral…)?\""}
```

Reply:
This sound good. Updated it.